platform/src/main/java/org/hillview/maps/LinearProjectionMap.java

```diff
@@ -0,0 +1,67 @@
+package org.hillview.maps;
+
+import org.hillview.dataset.api.IMap;
+import org.hillview.table.*;
+import org.hillview.table.api.*;
+import org.jblas.DoubleMatrix;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This map takes a list of column names and a projection matrix and applies the projection matrix to the matrix that is
+ * constructed from the table by horizontally stacking the specified columns. The resulting table is a copy of the old
+ * table, with the additional projected columns added to it. The new columns are named 'LinearProjection{i}'.
```

Comment:
I think that the names of the new columns should be members; with this algorithm you cannot run the LinearProjection twice on the same data.